libs/payments/stripe/src/lib/accountCustomer/accountCustomer.repository.spec.ts

```diff
@@ -0,0 +1,168 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+// import { faker } from '@faker-js/faker';
+import { Kysely } from 'kysely';
+
+import {
+  // AccountFactory,
+  AccountCustomerFactory,
+  DB,
+  testAccountDatabaseSetup,
+} from '@fxa/shared/db/mysql/account';
+
+// import { AccountCustomerNotUpdatedError } from './accountCustomer.error';
+import {
+  // accountExists,
+  createAccountCustomer,
+  deleteAccountCustomer,
+  getAccountCustomerByUid,
+  // getUidAndEmailByStripeCustomerId,
+  // updateAccountCustomer,
+} from './accountCustomer.repository';
+
+describe('AccountCustomer Repository', () => {
+  let kyselyDb: Kysely<DB>;
+
+  beforeAll(async () => {
+    kyselyDb = await testAccountDatabaseSetup(['accounts', 'accountCustomers']);
+  });
+
+  afterAll(async () => {
+    if (kyselyDb) {
+      await kyselyDb.destroy();
+    }
+  });
+
+  // describe('accountExists', () => {
+  // });
+
+  describe('createAccountCustomer', () => {
+    it('creates an accountCustomer in the DB when the uid and customer id are valid', async () => {
+      const mockAccountCustomer = AccountCustomerFactory();
+      const result = await createAccountCustomer(kyselyDb, mockAccountCustomer);
+
+      expect(result).toEqual(mockAccountCustomer);
+    });
+
+    it('fails to create an accountCustomer with a duplicateID in the DB', async () => {
+      const accountCustomer = AccountCustomerFactory();
+      await createAccountCustomer(kyselyDb, accountCustomer);
+      expect(
+        createAccountCustomer(kyselyDb, accountCustomer)
+      ).rejects.toThrow();
```

Comment:
Q: Previously we used something like:
```js
rejects.toBeInstanceOf(AccountCustomerNotDeletedError);
```

Should we be checking a specific instance vs general error throwness?

Reply:
I believe in `accountCustomer.manager.spec.ts`, but not in `accountCustomer.repository.spec.ts`.